qutebrowser/mainwindow/tabwidget.py

```diff
@@ -517,7 +517,9 @@ def tabSizeHint(self, index):
 
             no_pinned_count = self.count() - self.pinned_count
             pinned_width = tab_width_pinned_conf * self.pinned_count
-            no_pinned_width = self.width() - pinned_width
+            # Prevent any tabs from being smaller than the min size
+            no_pinned_width = max(self.width() - pinned_width,
+                                  minimum_size.width() * no_pinned_count)
```

Comment:
How does this interact with the `elif self.count() * minimum_size.width() > self.width():` above? I think the real bug is that that calculation doesn't account for pinned tabs, no? However, I *think* this should be equivalent, so we could remove that `elif` above maybe?